### PR TITLE
For #5741: Add option to close sessions selectively.

### PR DESCRIPTION
**Pull request by mcarare**
Refactor TabsAdapter to implement ListAdapter.
Refactor TabViewHolder.
Remove state observing and pass initial tab list:
The tab list and the selected tab do not change while fragment is displayed.

### CODE REVIEW

**Comment by mcarare** [comment]
This is one of the most requested features in Playstore reviews.

Added @jeffreygee to review the UI implementation (it is based on the current design with the addition of the close button, the standard one from AC also used in Fenix)

Adding @channingcarter for the go-ahead on adding this feature.

Below are screenshots in both themes and a video demo of the feature.

 

https://user-images.githubusercontent.com/48995920/143579024-aa254140-3584-4cee-a4d6-4c9b6a29b9db.mp4

File: app/src/main/java/org/mozilla/focus/session/ui/TabsAdapter.kt [hunk 1]
```diff
@@ -6,47 +6,35 @@ package org.mozilla.focus.session.ui
...
+            isCurrentSession = isCurrentSession.invoke(currentItem),
+            selectSession = selectSession,
+            closeSession = closeSession,
+        )
     }
 
-    private fun onUpdate(tabs: List<TabSessionState>) {
-        this.tabs = tabs
-        notifyDataSetChanged()
-    }
```
Comment (pocmo) [inline app/src/main/java/org/mozilla/focus/session/ui/TabsAdapter.kt:51]: There are a lot of changes, so it's not immediately clear to me: The old code seems to have updated tabs dynamically, while this new code uses a fixed list, why is that?

Ah, now I think I understand, the UI gets closed and therefore we do not need to update it, right?

**Review by pocmo — APPROVED** [review]
LGTM. Tested on my device too. 👍

File: app/src/main/java/org/mozilla/focus/session/ui/TabsAdapter.kt (hunk 1 above)
Comment (mcarare) [reply app/src/main/java/org/mozilla/focus/session/ui/TabsAdapter.kt:51]: Yes, the updating never happens while the list is displayed, so there is no need to observe changes.

**Comment by mcarare** [comment]
PR update: Updated UI test to reflect new view hierarchy.

**Pull request merged**